setup.py

```diff
@@ -65,6 +65,7 @@ def find_packages(path, base=""):
 deps_required = []
 
 setup(name=name,
+      python_requires='>=3.6',
```

Comment:
Just a query, whilst version pinning should allow us avoid problems in packages which depend upon this lib for py2.7, the additional requirement for py3.6+ means even dependent py3 packages won't work by default on Ubuntu Xenial (for example). Given Xenial's LTS status we may not be able to depend upon v2.0 in other open source projects for up to another two years.